dart/lib/src/utils/stack_trace_utils.dart

```diff
@@ -0,0 +1,25 @@
+import 'package:meta/meta.dart';
+
+@internal
+extension StackTraceUtils on String {
+  bool isStackTrace() {
+    final frameNumberPrefix = RegExp(r'^#\d+', multiLine: true);
```

Comment:
Regex creation is CPU expensive, ideally, we lazily create it and reuse it for the whole app's lifecycle.